### PR TITLE
feat: Add DAPO dataset and Deepseek-v3 config

### PR DESCRIPTION
**Pull request by yfw**
# What does this PR do ?

This PR pulls in the DAPO dataset from https://github.com/NVIDIA-NeMo/RL/pull/602 and also adds a config for running Deepseek-V3 on this dataset.

Closes https://github.com/NVIDIA-NeMo/RL/issues/1142

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information

Sample run with Deepseek-V3 training on DAPO-Math-17k and validation on AIME-2024:

Screenshot 2025-10-06 at 1 13 05 PM

## Summary by CodeRabbit

- New Features
  - Added DAPO Math17K dataset support for training and validation.
  - Introduced optional DAPO math answer verification with boxed-answer and normalization logic.
  - Enabled GRPO training setups for DAPO Math17K with Megatron and vLLM generation, including precision, batching, and sequence packing options.

- Configuration
  - Provided end-to-end example configs covering optimizer/scheduler defaults, parallelism, and resource settings.
  - Included logging integrations (W&B, TensorBoard) and environment/cluster parameters.
  - Added dataset, generation, and validation scheduling controls, plus checkpointing options.

### CODE REVIEW

File: nemo_rl/environments/math_environment.py [hunk 1]
```diff
@@ -74,6 +75,7 @@ def verify(
         pred_responses: list[str],
         ground_truths: list[str],
         return_extracted_answer: bool = False,
+        use_dapo_math_verifier: bool = False,
```
Comment (terrykong) [inline nemo_rl/environments/math_environment.py]: nit: what about using a string matching for `"hf_math_verify"` and `"dapo_math_verify"` and error otherwise. that way we don't create too many input args

File: examples/configs/grpo_dapomath17k_dsv3_megatron.yaml
Comment (terrykong) [inline examples/configs/grpo_dapomath17k_dsv3_megatron.yaml:1]: can we add these two configs to recipes, so that they have associated nightly tests? the dsv3 config is so large, we probably can't run it nightly. can you add it to the release.txt tests?

File: nemo_rl/environments/math_environment.py (hunk 1 above)
Comment (yfw) [reply nemo_rl/environments/math_environment.py]: Addressed in 00294d9baabcf4c4a7bf4aee138e41914c6cb94c

File: tests/test_suites/llm/grpo_dapomath17k_dsv3_megatron.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+NUM_NODES=32
+STEPS_PER_RUN=10
+MAX_STEPS=10
+NUM_RUNS=$(( (MAX_STEPS + STEPS_PER_RUN - 1) / STEPS_PER_RUN ))  # Round up
+NUM_MINUTES=240
+# ===== END CONFIG =====
+
+exit_if_max_steps_reached
+
+# Use checkpoint created from the 8K checkpoint in grpo-deepscaler-1.5b-8K.sh
```
Comment (terrykong) [inline tests/test_suites/llm/grpo_dapomath17k_dsv3_megatron.sh]: @yfw could you update this comment?

Comment (yfw) [reply tests/test_suites/llm/grpo_dapomath17k_dsv3_megatron.sh]: Resolved in dab0953dd15fdd7e3e4c04e3eeae1df760c2f97d

**Review by terrykong — APPROVED** [review]
